### PR TITLE
add kubernetes 1.21 support

### PR DESCRIPTION
**Pull request by raphaelauv**
#19555

### CODE REVIEW

**Comment by potiuk** [comment]
We should also update those version in README and docs - there are couple of places to do so. 

Also we should remove both 1.18 and 1.19: 

Here is our poluicy: https://github.com/apache/airflow/blob/main/README.md#support-for-python-and-kubernetes-versions

K8S 1.18 EOL was 4 months ago, 1.19 EOL was 2 weeks ago - via: https://endoflife.date/kubernetes

File: chart/values.schema.json [hunk 1]
```diff
@@ -132,7 +132,7 @@
                             "default": "/"
                         },
                         "pathType": {
-                            "description": "The pathType for the web Ingress (required for Kubernetes 1.19 and above).",
+                            "description": "The pathType for the web Ingress (required for Kubernetes 1.20 and above).",
```
Comment (jedcunningham) [inline chart/values.schema.json]: We should probably leave these as-is, or remove them altogether.

Comment (raphaelauv) [reply chart/values.schema.json]: ok I revert my change

**Review by potiuk — APPROVED** [review]

File: scripts/ci/libraries/_initialization.sh [hunk 2]
```diff
@@ -511,7 +511,7 @@ function initialization::initialize_provider_package_building() {
 # Determine versions of kubernetes cluster and tools used
 function initialization::initialize_kubernetes_variables() {
     # Currently supported versions of Kubernetes
-    CURRENT_KUBERNETES_VERSIONS+=("v1.20.2" "v1.19.7" "v1.18.15")
+    CURRENT_KUBERNETES_VERSIONS+=("v1.21.1" "v1.20.2")
```
Comment (jedcunningham) [inline scripts/ci/libraries/_initialization.sh:490]: @potiuk, should these be flipped so the default is the oldest?

Comment (potiuk) [reply scripts/ci/libraries/_initialization.sh:490]: Yes. That's indeed better idea.

Comment (raphaelauv) [reply scripts/ci/libraries/_initialization.sh:490]: done

**Review by jedcunningham — APPROVED** [review]

File: scripts/ci/libraries/_initialization.sh [hunk 3]
```diff
@@ -511,7 +511,7 @@ function initialization::initialize_provider_package_building() {
 # Determine versions of kubernetes cluster and tools used
 function initialization::initialize_kubernetes_variables() {
     # Currently supported versions of Kubernetes
-    CURRENT_KUBERNETES_VERSIONS+=("v1.20.2" "v1.19.7" "v1.18.15")
+    CURRENT_KUBERNETES_VERSIONS+=("v1.20.2" "v1.21.1")
```
Comment (jedcunningham) [inline scripts/ci/libraries/_initialization.sh]: The order needs to be flipped here too.

Comment (raphaelauv) [reply scripts/ci/libraries/_initialization.sh]: done

**Comment by potiuk** [comment]
Some intermittent problems that we need to fix. LGTM!

**Review by potiuk — APPROVED** [review]

**Pull request merged**